src/cascadia/Remoting/WindowManager.cpp

```
@@ -118,78 +118,60 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
+                //  They hopefully just died here. That's okay, let's just go
+                // ask the next in the line of succession. At the very worst,
+                // we'll find _us_, (likely last in the line).
 
-                        // In WindowManager::ProposeCommandline, had we been the
-                        // king originally, we would have started by setting
-                        // this to true. We became the monarch here, so set it
-                        // here as well.
-                        _shouldCreateWindow = true;
-                        return;
-                    }
+                // If the monarch (maybe us) failed for _any other reason_ than
+                // them dying. This IS quite unexpected. Let this bubble out.
+                TraceLoggingWrite(g_hRemotingProvider,
+                                  "WindowManager_proposeToMonarch_unexpectedExceptionFromKing",
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                                  TraceLoggingKeyword(TIL_KEYWORD_TRACE));
 
-                    // Here, we created the new monarch, it wasn't us, so we're
-                    // gonna go through the while loop again and ask the new
-                    // king.
-                    TraceLoggingWrite(g_hRemotingProvider,
-                                      "WindowManager_proposeToMonarch_tryAgain",
-                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
-                                      TraceLoggingKeyword(TIL_KEYWORD_TRACE));
-                }
-                else
+                LOG_CAUGHT_EXCEPTION();
+                // We failed to ask the monarch. It must have died. Try and
+                // find the real monarch. Don't perform an election, that
+                // assumes we have a peasant, which we don't yet.
+                _createMonarchAndCallbacks();

```

So I think the thing is... if there's a bad monarch wedged in registration, every time we hit this and eat an exception, we're going to look it up again from the registration catalog _and get the same object._

We had an explicit failout case before that would stop infinite retry... should we have one here?